### PR TITLE
Marge_Pr_001

### PR DESCRIPTION
**Pull request by DevStrikerTech**
1. Added User and Profile models.
2. Added testcases for both user and profile and passed successfully.

image

3. added paths for static, template and media files.
4. Runtime

image

### CODE REVIEW

File: backend/userauths/models.py [hunk 1]
```diff
@@ -1,3 +1,75 @@
...
+        """_summary_
+        """
+        email_username, full_name = self.email.split("@")
+        if self.full_name == "" or self.full_name == None:
+            self.full_name == email_username
+        if self.username == "" or self.username == None:
+            self.username = email_username
+        super(User, self).save(*args, **kwargs)
+   
+class Profile(models.Model):
```
Comment (MaveDataHolic) [inline backend/userauths/models.py:32]: Your class will only create user but not profile! Example you may have 500 users in the future and will you be adding those 500 users manually? Make use of **django.core.signals** to define new function that will handle profile creation and storing.

File: backend/userauths/models.py [hunk 2]
```diff
@@ -1,3 +1,75 @@
 from django.db import models
+from django.contrib.auth.models import AbstractUser
+
+class User(AbstractUser):
+    """_summary_
```
Comment (MaveDataHolic) [inline backend/userauths/models.py]: Why doc string is not having proper summary?

File: backend/userauths/models.py [hunk 3]
```diff
@@ -1,3 +1,75 @@
...
+    refresh_token = models.CharField(max_length=1000, null=True, blank=True)
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['username']
+
+    def __str__(self):
+        """_summary_
+
+        Returns:
+            _type_: _description_
```
Comment (MaveDataHolic) [inline backend/userauths/models.py]: Again return type doc string is not written properly!

**Review by MaveDataHolic — CHANGES_REQUESTED** [review]
Please make the changes and rest LGTM!

**Comment by MaveDataHolic** [comment]
Some more minor changes are required:
1. You are only providing 2 testcases so far, where is the rest or test coverage?
2. Showing local build is not suffice here, we are not going sent your local computer to cloud! 

Create code coverage and GitHub action for your workflow to show build for future.

**Comment by DevStrikerTech** [comment]
image

image

**Comment by NormCoderAcademy** [comment]
LGTM!

**Pull request merged**